frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -36,6 +29,9 @@ const defaultFormData = {
     roleId: null as number | null,
 };
 const formData = ref(JSON.parse(JSON.stringify(defaultFormData))); // clone default input
+const regularUserRoleOptions = computed(() => {
+    return FamLoginUserState.getCachedAppRoles(selectedApplicationId.value!);
```

Comment:
Hi Jota, we might can't reuse the existing `getCachedAppRoles` method here. Because the current `getCachedAppRoles` is getting roles for app admin, which can have access to all roles. But in our case here, the user could just be the delegated admin of the selected application, and it's possible he can only manage a few roles, not all. Sorry I missed this part as well. So I think we need to first check if user is the app admin or delegated admin of the selected app, and then we decide to get roles from auth_key "APP_ADMIN" or "DELEGATED_ADMIN". 

So basically you can call the `isAdminOfSelectedApplication` method first, and if return true, then we keep what we have, and if false, then the user will be delegated admin. (Cause as long as he can select that app, he will be app admin or delegated admin of it.) And then you just create a new method `getCachedAppRolesForDelegatedAdmin` and call it

<img width="1015" alt="image" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/b0ab14f1-d6ab-45a9-998b-b59500bdc2d2">

Reply:
Hello, Catherine! I'll make the change. Thanks!